src/core/contracts/contractErrors.ts

```diff
@@ -0,0 +1,49 @@
+import { DSNPUserId } from "../identifiers";
+import { DSNPError } from "../errors";
+
+/**
+ * ContractError indicates that an error occurred in the contracts module.
+ */
+export class ContractError extends DSNPError {
+  constructor(message: string) {
+    super(message);
+  }
+}
+
+/**
+ * MissingContractAddressError indicates that a contract address could not be
+ * found either on the chain or in the configuration overrides.
+ */
+export class MissingContractAddressError extends ContractError {
+  contractName: string;
+
+  constructor(contractName: string) {
+    super(`Could not find address for ${contractName} contract.`);
+    this.contractName = contractName;
+  }
+}
+
+/**
+ * MissingRegistrationContractError indicates that the registration for a given
+ * DSNP User Id could not be found.
+ */
+export class MissingRegistrationContractError extends ContractError {
```

Comment:
I was confused by this name. Perhaps drop `contract`?
```suggestion
export class MissingRegistrationError extends ContractError {
```